SessionMessagingKit/Jobs/JobQueue.swift

```diff
@@ -14,7 +16,15 @@ public final class JobQueue : NSObject, JobDelegate {
     }
 
     @objc public func addWithoutExecuting(_ job: Job, using transaction: Any) {
-        job.id = String(NSDate.millisecondTimestamp())
+        let timestamp = NSDate.millisecondTimestamp()
+        let count = JobQueue.jobIDs[timestamp] ?? 0
+        // When adding multiple jobs in rapid succession, timestamps might not be good enough as a unique ID. To
+        // deal with this we keep track of the number of jobs with a given timestamp and that to the end of the
+        // timestamp to make it a unique ID. We can't use a random number because we do still want to keep track
+        // of the order in which the jobs were added.
+        let id = String(timestamp) + String(count)
```

Comment:
String concatenation is fine for this? if 20 jobs added with a timestamp of "30" for instance you'll get "300", "301", ... "3019" or something but guessing that doesn't really matter as long as there's some ordering

Reply:
Yeah, the idea is that this way we preserve the order in which they were added (which is needed so that we can resume them in order if the user closes the app)